src/hash.cpp

```diff
@@ -54,12 +54,21 @@ namespace azure {  namespace storage_lite {
         unsigned char digest[SHA256_DIGEST_LENGTH];
 
 #ifdef USE_OPENSSL
+#if OPENSSL_VERSION_NUMBER < 0x10100000L
```

Comment:
Have you rebased to the latest commit? I think this code has already been merged.